### PR TITLE
begin moving indirect model imports out of __init__.py

### PR DESCRIPTION
**Pull request by nickpell**
Needed for django 1.9, which fails hard if models.py is imported directly or indirectly in **init**.py

Doing a first pass with a5288.

@dannyroberts @sravfeyn

### CODE REVIEW

**Comment by dannyroberts** [comment]
Related?

```
ImportError: No module named app-manager-testviews-domain
```
```
ImportError: No module named test-access-restrictions-domain
```
```
ImportError: No module named fandago
```

**Comment by nickpell** [comment]
✅

File: corehq/apps/reports/dispatcher.py [hunk 1]
```diff
@@ -84,7 +87,19 @@ def process(reports):
...
+
+        module_name = get_domain_module_map().get(domain)
+        if module_name is None:
+            custom_reports = ()
+        else:
+            module = __import__(module_name, fromlist=['reports'])
+            reports = getattr(module, 'reports')
+            custom_reports = process(getattr(reports, attr_name, ()))
+
+            # soon to be removed
```
Comment (emord) [inline corehq/apps/reports/dispatcher.py]: @nickpell I think that this can be removed now right?

Comment (nickpell) [reply corehq/apps/reports/dispatcher.py]: Looks like yes.

**Pull request merged**